pkg/operator/ceph/csi/version.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csi
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+
+	"github.com/pkg/errors"
+)
+
+// CephCSIVersion represents the Ceph CSI version format
+type CephCSIVersion struct {
+	Major  int
+	Minor  int
+	Bugfix int
+}
+
+var (
+	//minimum supported version is 2.0.0
+	minimum = CephCSIVersion{2, 0, 0}
+
+	//supportedCSIVersions are versions that rook supports
+	supportedCSIVersions = []CephCSIVersion{minimum}
+
+	// for parsing the output of `cephcsi`
+	versionCSIPattern = regexp.MustCompile(`v(\d+)\.(\d+)\.(\d+)`)
+)
+
+func (v *CephCSIVersion) String() string {
+	return fmt.Sprintf("%d.%d.%d",
+		v.Major, v.Minor, v.Bugfix)
+}
+
+// Supported checks if the detected version is part of the known supported CSI versions
+func (v *CephCSIVersion) Supported() bool {
+	for _, sv := range supportedCSIVersions {
```

Comment:
What about this approach?
```
If less than the min version
  return false
If AllowUnsupported
  return true
for each supported version
  if major and minor versions equal
    return true
```

Reply:
You could also re-use the [IsSuperior()](https://github.com/rook/rook/blob/27a02386c353d416e7a61ede8b08a6d4ddde9819/pkg/operator/ceph/version/version.go#L208) method from the ceph version checking, at least for the min version comparison.

Reply:
@travisn why not use `v.Major == sv.Major && v.Minor == sv.Minor && v.Bugfix >= sv.Bugfix` to avoid multiple if blocks here?

Reply:
In my opinion the is code is easier to read and to understand when using multiple blocks instead of a lengthy concatenation of Boolean expressions.

Reply:
I'm fine either way